### PR TITLE
feat(alerting): magic alerting severities

### PR DESCRIPTION
**Pull request by fourstepper**

### CODE REVIEW

**Review by Hy3n4 — APPROVED** [review]
LGTM

File: internal/helpers/prometheus_helper.go [hunk 1]
```diff
@@ -327,11 +327,13 @@ func (mrs *MonitoringRuleSet) SetupRules() ([]monitoringv1.RuleGroup, error) {
...
 
+	// TODO: having the magic alerting we still need to figure out how to pass the severity into the alerting rule when we have more than one alerting tool.
+	// The idea is to have a map of the alerting tool and the severity and then iterate over all connected AlertNotificationTargets.
+
 	if mrs.Slo.ObjectMeta.Annotations["osko.dev/magicAlerting"] == "true" {
 		duration := monitoringv1.Duration("5m")
 		var alertRules []monitoringv1.Rule
-		alertRules = append(alertRules, mrs.createMagicMultiBurnRateAlert(alertRuleErrorBudgets, "0.001", &duration, "page"))
-		alertRules = append(alertRules, mrs.createMagicMultiBurnRateAlert(alertRuleErrorBudgets, "0.001", &duration, "ticket"))
+		alertRules = append(alertRules, mrs.createMagicMultiBurnRateAlert(alertRuleErrorBudgets, "0.001", &duration, config.Cfg.AlertSeverities.HighFast))
```
Comment (fourstepper) [inline internal/helpers/prometheus_helper.go:336]: This probably shouldn't be merged like this @Hy3n4

**Comment by fourstepper** [comment]
`func (mrs *MonitoringRuleSet) createMagicMultiBurnRateAlert(burnRates []monitoringv1.Rule, threshold string, duration *monitoringv1.Duration, severity string)`

change `severity string` to `config/types.go` `AlertSeverities`

File: internal/helpers/prometheus_helper.go (hunk 1 above)
Comment (fourstepper) [reply internal/helpers/prometheus_helper.go:336]: `func (mrs *MonitoringRuleSet) createMagicMultiBurnRateAlert(burnRates []monitoringv1.Rule, threshold string, duration *monitoringv1.Duration, severity string)`

change `severity string` to `config/types.go` `AlertSeverities`

File: internal/helpers/prometheus_helper.go [hunk 2]
```diff
@@ -368,18 +369,18 @@ func (mrs *MonitoringRuleSet) createMagicMultiBurnRateAlert(burnRates []monitori
 	if severity == "page" {
```
Comment (fourstepper) [inline internal/helpers/prometheus_helper.go:369]: We construct both severity page and ticket in this one function instead of this `if else if magic`

**Comment by fourstepper** [comment]
Closes https://github.com/oskoperator/osko/issues/103

File: internal/helpers/prometheus_helper.go [hunk 3]
```diff
@@ -258,7 +260,19 @@ func (mrs *MonitoringRuleSet) SetupRules() ([]monitoringv1.RuleGroup, error) {
...
+
+	var uniqueWindows []string
+	for _, w := range windows {
+		if !windowSet[w] {
+			windowSet[w] = true
+			uniqueWindows = append(uniqueWindows, w)
+		}
+	}
+
+	log.V(1).Info("Windows after deduplication", "uniqueWindows", uniqueWindows)
```
Comment (fourstepper) [inline internal/helpers/prometheus_helper.go:274]: https://go.dev/play/p/2vsuJaDuyAb

a bit simpler :D

File: internal/helpers/prometheus_helper.go [hunk 4]
```diff
@@ -280,12 +294,19 @@ func (mrs *MonitoringRuleSet) SetupRules() ([]monitoringv1.RuleGroup, error) {
 	rules["errorBudgetValue"][baseWindow] = mrs.createErrorBudgetValueRecordingRule(rules["sliMeasurement"][baseWindow], baseWindow)
 	rules["errorBudgetTarget"][baseWindow] = mrs.createErrorBudgetTargetRecordingRule(baseWindow)
 	rules["burnRate"][baseWindow] = mrs.createBurnRateRecordingRule(rules["errorBudgetValue"][baseWindow], rules["errorBudgetTarget"][baseWindow], baseWindow)
+	if baseWindow == "5m" || baseWindow == "30m" || baseWindow == "1h" || baseWindow == "6h" ||
+		baseWindow == "24h" || baseWindow == "3d" || baseWindow == "2h" {
+		alertRuleErrorBudgets = append(alertRuleErrorBudgets, rules["burnRate"][baseWindow])
+	}
```
Comment (fourstepper) [inline internal/helpers/prometheus_helper.go:300]: what are we trying to do with the if?

**Pull request merged**